src/HydrationMiddleware/HydratePublicProperties.php

```diff
@@ -94,6 +97,10 @@ public static function dehydrate($instance, $response)
                 is_bool($value) || is_null($value) || is_array($value) || is_numeric($value) || is_string($value)
             ) {
                 data_set($response, 'memo.data.'.$key, $value);
+            } else if (version_compare(PHP_VERSION, '7.4', '>=') && LivewireProperty::has($value)) {
+                $response->memo['dataMeta']['customProperties'][] = $key;
+
+                data_set($response, 'memo.data.'.$key, LivewireProperty::dehydrate($value));
```

Comment:
This is only checking the value type to determine if it's a custom property. This means that the following scenario would not work:
```php
public ?Carbon $cancellationDate = null;
```
```blade
<x-input.date wire:model="cancellationDate" />
```

Ideally a custom property should be allowed to be `null` and Livewire should check the defined type using Reflection to know if it's a custom property. The value type is really a last resort if the Reflection check fails.